src/plugin/BodyROSItem.cpp

```diff
@@ -455,7 +455,61 @@ void BodyROSItem::updateVisionSensor
     vision.step = sensor->image().width() * sensor->image().numComponents();
     vision.data.resize(vision.step * vision.height);
     std::memcpy(&(vision.data[0]), &(sensor->image().pixels()[0]), vision.step * vision.height);
-    publisher.publish(vision);
+
+    sensor_msgs::CameraInfo info;
+    info.header.stamp = vision.header.stamp;
+    info.header.frame_id  = vision.header.frame_id ;
```

Comment:
Each ROS msg type havs a copy constructor, so you can copy a msg as 
```suggestion
    info.header = vision.header;
```